src/sas/qtgui/Utilities/MuMag/MuMag.py

```diff
@@ -0,0 +1,336 @@
+
+    def importData(self):
+        """ Callback for the import data button """
+
+        # Get the directory from the user
+        directory = MuMagLib.directory_popup()
+
+        if directory is None:
+            log.info("No directory selected")
+            return
+
+        try:
+            self.data = MuMagLib.import_data(directory)
+        except LoadFailure as lf:
+            log.error(repr(lf))
+            return
+
+        log.info(f"Loaded {len(self.data)} datasets")
+
+        self.hide_everything()
+        self.plot_tabs.setTabEnabled(0, True)
+        self.show_input_data()
+
+    def hide_everything(self):
+        """ Hide all plots, disable tabs"""
+
+        self.data_axes.cla()
+        self.chi_squared_axes.cla()
+        self.residual_axes.cla()
+        self.s_h_axes.cla()
+        self.longitudinal_scattering_axes.cla()
+
+        self.plot_tabs.setTabEnabled(1, False)
+        self.plot_tabs.setTabEnabled(2, False)
+        # weird order because of how the widget behaves when all are not enabled
+        self.plot_tabs.setTabEnabled(0, False)
+
+
+    def show_input_data(self):
+        """ Plot Experimental Data: Generate Figure """
+
+        colors = pl.cm.jet(np.linspace(0, 1, len(self.data)))
+
+        for i, datum in enumerate(self.data):
+
+            self.data_axes.loglog(datum.scattering_curve.x,
+                      datum.scattering_curve.y,
+                      linestyle='-', color=colors[i], linewidth=0.5,
+                      label=r'$B_0 = ' + str(datum.applied_field) + '$ T')
+
+            self.data_axes.loglog(datum.scattering_curve.x,
+                      datum.scattering_curve.y, '.',
+                      color=colors[i], linewidth=0.3, markersize=1)
+
+        # Plot limits
+        qlim = MuMagLib.nice_log_plot_bounds([datum.scattering_curve.x for datum in self.data])
+        ilim = MuMagLib.nice_log_plot_bounds([datum.scattering_curve.y for datum in self.data])
+
+        self.data_axes.set_xlabel(r'$q$ [1/nm]')
+        self.data_axes.set_ylabel(r'$I_{\mathrm{exp}}$')
+        self.data_axes.set_xlim(qlim)
+        self.data_axes.set_ylim(ilim)
+        self.data_figure.tight_layout()
+
+        self.data_axes.set_visible(True)
+        self.data_figure.canvas.draw()
+
+
+    def get_fit_parameters(self) -> FitParameters:
+        """ Get an object containing all the parameters needed for doing the fitting """
+
+        a_min = self.aMinSpinBox.value()
+        a_max = self.aMaxSpinBox.value()
+
+        if a_max <= a_min:
+            raise ValueError(f"minimum A must be less than maximum A")
+
+        match self.ScatteringGeometrySelect.currentText().lower():
+            case "parallel":
+                geometry = ExperimentGeometry.PARALLEL
+            case "perpendicular":
+                geometry = ExperimentGeometry.PERPENDICULAR
+            case _:
+                raise ValueError(f"Unknown experiment geometry: {self.ScatteringGeometrySelect.currentText()}")
```

Comment:
To prevent this from growing too large if future geometries are added:
```suggestion
        try:
            geometry = ExperimentGeometry[self.ScatteringGeometrySelect.currentText().upper()]
        except KeyError:
            raise ValueError(f"Unknown experiment geometry: {self.ScatteringGeometrySelect.currentText()}")
```

Reply:
I disagree, on the basis of readability.